tests/functional/aws-node-sdk/test/object/websiteGet.js

```diff
@@ -576,5 +577,230 @@ describe('User visits bucket website endpoint', () => {
                 });
             });
         });
+
+        describe('with key x-amz-website-redirect-location header ' +
+        'to /redirect.html', () => {
+            beforeEach(() => {
+                const webConfig = new WebsiteConfigTester('index.html');
+                return s3.putBucketWebsiteAsync({ Bucket: bucket,
+                    WebsiteConfiguration: webConfig })
+                .then(() => s3.putObjectAsync({ Bucket: bucket,
+                    Key: 'index.html',
+                    ACL: 'public-read',
+                    Body: fs.readFileSync(path.join(__dirname,
+                        '/websiteFiles/index.html')),
+                    ContentType: 'text/html',
+                    WebsiteRedirectLocation: '/redirect.html' }))
+                .then(() => s3.putObjectAsync({ Bucket: bucket,
+                    Key: 'redirect.html',
+                    ACL: 'public-read',
+                    Body: fs.readFileSync(path.join(__dirname,
+                        '/websiteFiles/redirect.html')),
+                    ContentType: 'text/html' }));
+            });
+
+            afterEach(() => bucketUtil.empty(bucket));
+
+            it('should serve redirect file', done => {
+                browser.visit(`${endpoint}`, () => {
+                    WebsiteConfigTester.checkHTML(browser, 'redirect-user');
+                    done();
+                });
+            });
+        });
+
+        describe('with key x-amz-website-redirect-location header ' +
+        'to another website', () => {
+            beforeEach(() => {
+                const webConfig = new WebsiteConfigTester('index.html');
+                return s3.putBucketWebsiteAsync({ Bucket: bucket,
+                    WebsiteConfiguration: webConfig })
+                .then(() => s3.putObjectAsync({ Bucket: bucket,
+                    Key: 'index.html',
+                    ACL: 'public-read',
+                    Body: fs.readFileSync(path.join(__dirname,
```

Comment:
Couldn't this be asyncronous? Or `Body` can take a streaming object
https://nodejs.org/api/fs.html#fs_fs_createreadstream_path_options